src/expression/logical/xor.ts

```diff
@@ -50,4 +53,37 @@ export class Xor extends Logical {
     }
     return res
   }
+
+  simplify (...args: [Context]): boolean | Evaluable {
+    const [evaluablesLeft, trueCount] = this.operands.reduce<[Evaluable[], number]>(
+      ([notSimplifiedConditions, trueCount], child) => {
+        if (trueCount > 1) {
+          return [notSimplifiedConditions, trueCount]
+        }
+        const childResult = child.simplify(...args)
+        if (!isBoolean(childResult)) {
+          return [[...notSimplifiedConditions, child], trueCount]
+        }
+        if (childResult) {
+          return [notSimplifiedConditions, trueCount + 1]
+        }
+        return [notSimplifiedConditions, trueCount]
+      }, [[], 0])
+    if (trueCount > 1) {
+      return false
+    }
+    if (evaluablesLeft.length === 0) {
+      return trueCount === 1
+    }
+    if (evaluablesLeft.length === 1) {
+      if (trueCount === 1) {
+        return new Not(...evaluablesLeft)
+      }
+      return evaluablesLeft[0]
+    }
+    if (trueCount === 1) {
+      return new Nor(evaluablesLeft)
+    }
+    return new Xor(evaluablesLeft)
```

Comment:
Do you not prefer whitespace for visual separation? Asking out of curiosity.  Not asking as a prompt to change this.

Reply:
I don't mind the whitespace. I prefer the code tight and small but up to you.

Reply:
Terse code.

No no.  I'm not going to request these sort of changes.  Everyone just needs to understand that code should be readable for both peers and yourself.

Reply:
Do you think the code is unreadable because there is not empty line after the if block?

Reply:
It is readable.